### ISSUE TITLE
tutorial 1: "incompatible versions in resolved dependencies" running bootstrap script

### ISSUE THREAD

**Issue by mnyrop** [inveniosoftware/training#80]
Hello,

In both the Vagrant setup and the OVA box, I'm unable to lock my dependencies when running  
`$ ./scripts/bootstrap`. There seems to be an issue with `invenio-base` ...?

The full response is reproduced below:

```sh
bootcamp@invenio:~/src/my-site$ ./scripts/bootstrap
'Pipfile.lock' not found. Generating via 'pipenv lock --dev'...
Creating a virtualenv for this project…
Pipfile: /home/bootcamp/src/my-site/Pipfile
Using /usr/bin/python3.6m (3.6.7) to create virtualenv…
⠹ Creating virtual environment...Using base prefix '/usr'
New python executable in /home/bootcamp/.local/share/virtualenvs/my-site-7Oi5HgLM/bin/python3.6m
Also creating executable in /home/bootcamp/.local/share/virtualenvs/my-site-7Oi5HgLM/bin/python
Installing setuptools, pip, wheel...
done.
Running virtualenv with interpreter /usr/bin/python3.6m

✔ Successfully created virtual environment! 
Virtualenv location: /home/bootcamp/.local/share/virtualenvs/my-site-7Oi5HgLM
Locking [dev-packages] dependencies…
✔ Success! 
Locking [packages] dependencies…
✘ Locking Failed! 
[pipenv.exceptions.ResolutionFailure]:       req_dir=requirements_dir
[pipenv.exceptions.ResolutionFailure]:   File "/home/bootcamp/.local/lib/python3.6/site-packages/pipenv/utils.py", line 726, in resolve_deps
[pipenv.exceptions.ResolutionFailure]:       req_dir=req_dir,
[pipenv.exceptions.ResolutionFailure]:   File "/home/bootcamp/.local/lib/python3.6/site-packages/pipenv/utils.py", line 480, in actually_resolve_deps
[pipenv.exceptions.ResolutionFailure]:       resolved_tree = resolver.resolve()
[pipenv.exceptions.ResolutionFailure]:   File "/home/bootcamp/.local/lib/python3.6/site-packages/pipenv/utils.py", line 395, in resolve
[pipenv.exceptions.ResolutionFailure]:       raise ResolutionFailure(message=str(e))
[pipenv.exceptions.ResolutionFailure]:       pipenv.exceptions.ResolutionFailure: ERROR: ERROR: Could not find a version that matches invenio-base<1.1.0,>=1.0.0,>=1.0.2,>=1.2.2
[pipenv.exceptions.ResolutionFailure]:       Tried: 0.1.0, 0.1.0, 0.2.0, 0.2.0, 0.2.1, 0.2.1, 0.3.0, 0.3.0, 0.3.1, 0.3.1, 1.0.0, 1.0.0, 1.0.1, 1.0.1, 1.0.2, 1.0.2, 1.1.0, 1.1.0, 1.2.0, 1.2.0, 1.2.1, 1.2.1, 1.2.2, 1.2.2, 1.2.3, 1.2.3
[pipenv.exceptions.ResolutionFailure]:       Skipped pre-versions: 1.0.0a1, 1.0.0a1, 1.0.0a2, 1.0.0a2, 1.0.0a3, 1.0.0a3, 1.0.0a4, 1.0.0a4, 1.0.0a5, 1.0.0a5, 1.0.0a6, 1.0.0a7, 1.0.0a7, 1.0.0a8, 1.0.0a8, 1.0.0a9, 1.0.0a9, 1.0.0a10, 1.0.0a10, 1.0.0a11, 1.0.0a12, 1.0.0a12, 1.0.0a13, 1.0.0a13, 1.0.0a14, 1.0.0a14, 1.0.0a15, 1.0.0a15, 1.0.0a16, 1.0.0a16, 1.0.0b1, 1.0.0b1
[pipenv.exceptions.ResolutionFailure]: Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  First try clearing your dependency cache with $ pipenv lock --clear, then try the original command again.
 Alternatively, you can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
ERROR: ERROR: Could not find a version that matches invenio-base<1.1.0,>=1.0.0,>=1.0.2,>=1.2.2
Tried: 0.1.0, 0.1.0, 0.2.0, 0.2.0, 0.2.1, 0.2.1, 0.3.0, 0.3.0, 0.3.1, 0.3.1, 1.0.0, 1.0.0, 1.0.1, 1.0.1, 1.0.2, 1.0.2, 1.1.0, 1.1.0, 1.2.0, 1.2.0, 1.2.1, 1.2.1, 1.2.2, 1.2.2, 1.2.3, 1.2.3
Skipped pre-versions: 1.0.0a1, 1.0.0a1, 1.0.0a2, 1.0.0a2, 1.0.0a3, 1.0.0a3, 1.0.0a4, 1.0.0a4, 1.0.0a5, 1.0.0a5, 1.0.0a6, 1.0.0a7, 1.0.0a7, 1.0.0a8, 1.0.0a8, 1.0.0a9, 1.0.0a9, 1.0.0a10, 1.0.0a10, 1.0.0a11, 1.0.0a12, 1.0.0a12, 1.0.0a13, 1.0.0a13, 1.0.0a14, 1.0.0a14, 1.0.0a15, 1.0.0a15, 1.0.0a16, 1.0.0a16, 1.0.0b1, 1.0.0b1
There are incompatible versions in the resolved dependencies.
[pipenv.exceptions.ResolutionFailure]:   File "/home/bootcamp/.local/lib/python3.6/site-packages/pipenv/utils.py", line 726, in resolve_deps
[pipenv.exceptions.ResolutionFailure]:       req_dir=req_dir,
[pipenv.exceptions.ResolutionFailure]:   File "/home/bootcamp/.local/lib/python3.6/site-packages/pipenv/utils.py", line 480, in actually_resolve_deps
[pipenv.exceptions.ResolutionFailure]:       resolved_tree = resolver.resolve()
[pipenv.exceptions.ResolutionFailure]:   File "/home/bootcamp/.local/lib/python3.6/site-packages/pipenv/utils.py", line 395, in resolve
[pipenv.exceptions.ResolutionFailure]:       raise ResolutionFailure(message=str(e))
[pipenv.exceptions.ResolutionFailure]:       pipenv.exceptions.ResolutionFailure: ERROR: ERROR: Could not find a version that matches invenio-base<1.1.0,>=1.0.0,>=1.0.2,>=1.2.2
[pipenv.exceptions.ResolutionFailure]:       Tried: 0.1.0, 0.1.0, 0.2.0, 0.2.0, 0.2.1, 0.2.1, 0.3.0, 0.3.0, 0.3.1, 0.3.1, 1.0.0, 1.0.0, 1.0.1, 1.0.1, 1.0.2, 1.0.2, 1.1.0, 1.1.0, 1.2.0, 1.2.0, 1.2.1, 1.2.1, 1.2.2, 1.2.2, 1.2.3, 1.2.3
[pipenv.exceptions.ResolutionFailure]:       Skipped pre-versions: 1.0.0a1, 1.0.0a1, 1.0.0a2, 1.0.0a2, 1.0.0a3, 1.0.0a3, 1.0.0a4, 1.0.0a4, 1.0.0a5, 1.0.0a5, 1.0.0a6, 1.0.0a7, 1.0.0a7, 1.0.0a8, 1.0.0a8, 1.0.0a9, 1.0.0a9, 1.0.0a10, 1.0.0a10, 1.0.0a11, 1.0.0a12, 1.0.0a12, 1.0.0a13, 1.0.0a13, 1.0.0a14, 1.0.0a14, 1.0.0a15, 1.0.0a15, 1.0.0a16, 1.0.0a16, 1.0.0b1, 1.0.0b1
[pipenv.exceptions.ResolutionFailure]: Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  First try clearing your dependency cache with $ pipenv lock --clear, then try the original command again.
 Alternatively, you can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
ERROR: ERROR: Could not find a version that matches invenio-base<1.1.0,>=1.0.0,>=1.0.2,>=1.2.2
Tried: 0.1.0, 0.1.0, 0.2.0, 0.2.0, 0.2.1, 0.2.1, 0.3.0, 0.3.0, 0.3.1, 0.3.1, 1.0.0, 1.0.0, 1.0.1, 1.0.1, 1.0.2, 1.0.2, 1.1.0, 1.1.0, 1.2.0, 1.2.0, 1.2.1, 1.2.1, 1.2.2, 1.2.2, 1.2.3, 1.2.3
Skipped pre-versions: 1.0.0a1, 1.0.0a1, 1.0.0a2, 1.0.0a2, 1.0.0a3, 1.0.0a3, 1.0.0a4, 1.0.0a4, 1.0.0a5, 1.0.0a5, 1.0.0a6, 1.0.0a7, 1.0.0a7, 1.0.0a8, 1.0.0a8, 1.0.0a9, 1.0.0a9, 1.0.0a10, 1.0.0a10, 1.0.0a11, 1.0.0a12, 1.0.0a12, 1.0.0a13, 1.0.0a13, 1.0.0a14, 1.0.0a14, 1.0.0a15, 1.0.0a15, 1.0.0a16, 1.0.0a16, 1.0.0b1, 1.0.0b1
There are incompatible versions in the resolved dependencies.
```

Any advice for how to proceed? Happy to share more info if useful.

**Comment by ppanero** [comment]
Hi @mnyrop Which branch/version was used to cookiecut the instance?

**Comment by mnyrop** [comment]
Thanks for responding @ppanero ! I'm following the instructions here: https://github.com/inveniosoftware/training/tree/master/01-getting-started#step-3-scaffold, so v3.2 is cookiecut

**Comment by ppanero** [comment]
Hy @mnyrop we have released v3.3.0, so using branch v3.3 should also work. I am closing the ticket but please do re-open it or open a new one if you encounter the same issue. Thanks!
